packages/core/src/context.ts

```diff
@@ -17,10 +20,24 @@ const logger = createLogger({
 let aiCoreServiceBinding: Service | undefined;
 
 /**
- * Returns a destination object from AI Core service binding.
+ * Returns a destination object.
+ * @param destination - The destination to use for the request.
  * @returns The destination object.
  */
-export async function getAiCoreDestination(): Promise<HttpDestination> {
+export async function getAiCoreDestination(
+  destination?: HttpDestinationOrFetchOptions
+): Promise<HttpDestination> {
+  // If Destination is provided, get the destination and return it.
+  if (destination) {
+    const resolvedDestination = await useOrFetchDestination(destination);
```

Comment:
[q] What about caching? If a name is passed and not the full destination, should the destination be cached?

Reply:
I mean they can decide whether to set `useCache` or not right? The default behaviour will be the same as cloud SDK.

```ts
DeploymentApi.deploymentQuery(queryParams, {
    'AI-Resource-Group': resourceGroup
  }).execute({
    destinationName: 'e2e-aicore',
    useCache: true
  });
```

Reply:
The default behavior in the SDK is to not cache, but this will change in v4 (would be a breaking change)

Reply:
> The default behavior in the SDK is to not cache, but this will change in v4 (would be a breaking change)

Yes, @jjtang1985 should we then cache by default in ai-sdk already?

Reply:
I will cache by default.